nginx.master.conf

```diff
@@ -285,35 +285,42 @@ http {
             proxy_pass http://log/;
         }
 
-        location ~ ^/system/v1/agent/(?<agentid>[0-9a-zA-Z-]+)/logs/v1/(?<url>.*)$ {
+        location /system/v1/metrics/ {
+            access_by_lua 'auth.validate_jwt_or_exit()';
+
+            include common/proxy-headers.conf;
+            proxy_pass http://metrics/;
+        }
+
+		# Ordering is important for these two endpoints!!!
+        location ~ ^/system/v1/agent/(?<agentid>[0-9a-zA-Z-]+)(?<type>(/logs/v1))(?<url>.*)$ {
             access_by_lua 'auth.validate_jwt_or_exit()';
+
             set $agentaddr '';
-            more_clear_input_headers Accept-Encoding;
-            rewrite ^/(slave|agent)/[0-9a-zA-Z-]+/.*$ $url break;
             rewrite_by_lua_file conf/master/agent.lua;
 
+            rewrite ^/agent/[0-9a-zA-Z-]+(.*)$ $1 break;
+
+            more_clear_input_headers Accept-Encoding;
```

Comment:
Does not block merge, but I'd really love to understand why we do this header modification. Probably worth a comment in one of the next iterations.